import "libnutrition/proto/data/nutrient.proto";
import "libnutrition/proto/service/error.proto";

message NutrientLoadRequest 
{
	optional bool omit = 1 [default = false];
	optional bool all = 2 [default = false];
	repeated string requestedIds = 3;
	repeated string requestedNames = 4;
	
	enum Category {
        	Energy = 0;
          	Basic = 1;
          	Vitamin = 2;
          	Mineral = 3;
          	Additional = 4;
          	Hidden = 5;
      	}
	
	repeated Category requestedCategories = 5;
}

message NutrientLoadResponse
{
	repeated NutrientData nutrients = 1;
	optional Error error = 100;
}

	